src/filesystem/implementations/as.h

```diff
@@ -406,17 +409,21 @@ ASFileSystem::DownloadFolder(
             "Failed to download file at " + blob_item.Name + ":" + ex.what());
       }
     }
-    for (const auto& directory_item : blob_prefixes) {
-      const auto& local_path = JoinPath({dest, BaseName(directory_item)});
-      int status = mkdir(
-          const_cast<char*>(local_path.c_str()), S_IRUSR | S_IWUSR | S_IXUSR);
-      if (status == -1) {
-        return Status(
-            Status::Code::INTERNAL,
-            "Failed to create local folder: " + local_path +
-                ", errno:" + strerror(errno));
+    if (recursive) {
+      for (const auto& directory_item : blob_prefixes) {
+        const auto& local_path = JoinPath({dest, BaseName(directory_item)});
+        int status = mkdir(
```

Comment:
Shouldn't the `FileSystem` wrapped API be used here as well?

Reply:
right, for Windows. good catch

Reply:
changed